File: 0202-happy-number/0202-happy-number.cs
public class Solution {
    public bool IsHappy(int n) {
        HashSet<int> hmap = new HashSet<int>();
        int num = n;
        while(true){
            int[] intArr = num.ToString().Select(digit => int.Parse(digit.ToString())).ToArray();
            int sum = 0;
            for(int i=0; i < intArr.Length; i ++){
                sum += (int)Math.Pow(intArr[i], 2);
            }
            
            if (sum == 1){ return true; }
            else if (hmap.Contains(sum)) { return false; }
            else{
                hmap.Add(sum);
                num = sum;
                continue;
            }
        }
        
    }
}

File: 0202-happy-number/0202-happy-number.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        hset = set()
        s = str(n)
        
        while s != "1":
            if s in hset:
                return False
            hset.add(s)
            ns = 0
            for char in s:
                ns += int(char) ** 2 
            s = str(ns)
        return True